components/application-connectivity-validator/internal/controller/cachesynccontroller.go

```diff
@@ -0,0 +1,216 @@
+package controller
+
+import (
+	"fmt"
+	"time"
+
+	gocache "github.com/patrickmn/go-cache"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+
+	"github.com/kyma-project/kyma/components/application-operator/pkg/apis/applicationconnector/v1alpha1"
+	clientset "github.com/kyma-project/kyma/components/application-operator/pkg/client/clientset/versioned"
+	log "github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/api/errors"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	informers "github.com/kyma-project/kyma/components/application-operator/pkg/client/informers/externalversions/applicationconnector/v1alpha1"
+	listers "github.com/kyma-project/kyma/components/application-operator/pkg/client/listers/applicationconnector/v1alpha1"
+)
+
+const controllerAgentName = "application-cache-controller"
+
+const (
+	// SuccessSynced is used as part of the Event 'reason' when a Application is synced
+	SuccessSynced = "Synced"
+	// ErrResourceExists is used as part of the Event 'reason' when a Application fails
+	// to sync due to a Deployment of the same name already existing.
+	ErrResourceExists = "ErrResourceExists"
+
+	// MessageResourceExists is the message used for Events when a resource
+	// fails to sync due to a Deployment already existing
+	MessageResourceExists = "Resource %q already exists and is not managed by Application"
+	// MessageResourceSynced is the message used for an Event fired when a Application
+	// is synced successfully
+	MessageResourceSynced = "Application resource synced successfully"
+)
+
+// Controller is the controller implementation for Application resources
+type Controller struct {
```

Comment:
At the moment controller caches all Applications CRs. While Validator is deployed by Application Operator per single Application CR, we don't really need to cache/watch all Applications.

Reply:
Yes, you are right. I am already aware of this and I already mentioned this in the description of the PR as a future improvement. I would like to merge this PR without this enhancement if it is okay? As a next step I will do this improvement by filtering events.

Reply:
Oh, sorry, I missed that. It would be great if you could also implement this in this PR.

Reply:
done. Please have a look.